targets/UnrealMarketplacePlugin/examplesource/ExampleProject/Plugins/PlayFab/Source/PlayFab/Private/Tests/PlayFabCppTests.cpp.ejs

```diff
@@ -8,62 +8,40 @@
 #include "Core/PlayFabServerAPI.h"
 #include "Core/PlayFabDataAPI.h"
 #include "Core/PlayFabAuthenticationAPI.h"
+#include "TestFramework/PlayFabTestRunner.h"
 
-void UPlayFabCppTests::ClassSetUp()
+void UPlayFabCppTests::SetTestTitleData(UTestTitleDataLoader& testTitleData)
 {
-    bool success = true;
-
-    FString jsonInput;
-    FString filename = TEST_TITLE_DATA_LOC;
-
-    if (!FPaths::FileExists(filename))
-    {
-        // Prefer to load path from environment variable, if present
-#if PLATFORM_WINDOWS
-        char* envPath = nullptr;
-        size_t envPathStrLen;
-        errno_t err = _dupenv_s(&envPath, &envPathStrLen, "PF_TEST_TITLE_DATA_JSON");
-        if (err == 0 && envPath != nullptr)
-            filename = FString(ANSI_TO_TCHAR(envPath));
-
-        if (envPath != nullptr)
-            free(envPath);
-#endif
-    }
-
-    success &= FFileHelper::LoadFileToString(jsonInput, *filename);
-
-    TSharedPtr<FJsonObject> jsonParsed = nullptr;
-    if (success)
-    {
-        TSharedRef<TJsonReader<>> jsonReader = TJsonReaderFactory<>::Create(jsonInput);
-        success &= FJsonSerializer::Deserialize(jsonReader, jsonParsed);
-    }
-
-    if (success) success &= jsonParsed->TryGetStringField("titleId", TitleId);
-    if (success) success &= jsonParsed->TryGetStringField("developerSecretKey", DevSecretKey);
-    if (success) success &= jsonParsed->TryGetStringField("userEmail", UserEmail);
+    UserEmail = testTitleData.userEmail;
+}
 
-    if (!success)
-        return;
+void UPlayFabCppTests::ClassSetUp()
+{
+    IPlayFab* playFabSettings = &(IPlayFab::Get());
+    playFabSettings->setAdvertisingIdType(TEXT(""));
+    playFabSettings->setAdvertisingIdValue(TEXT(""));
 
     ClientAPI = IPlayFabModuleInterface::Get().GetClientAPI();
     ServerAPI = IPlayFabModuleInterface::Get().GetServerAPI();
     DataAPI = IPlayFabModuleInterface::Get().GetDataAPI();
     AuthenticationAPI = IPlayFabModuleInterface::Get().GetAuthenticationAPI();
-
-    checkf(ClientAPI.IsValid(), TEXT("The ClientAPI reports itself as invalid."));
-    checkf(ServerAPI.IsValid(), TEXT("The ServerAPI reports itself as invalid."));
```

Comment:
Why are we removing validity checks for ClientAPI, DataAPI and AuthenticationAPI? 
I see that we still check ServerAPI validity during ::SetUp

Reply:
1. I can't imagine a scenario where one will work and the others wont.
We could theoretically test all 4, but I think it's unnecessary.
Also, I've never seen it happen. Given how it compiles, I don't think it's actually possible.

2. Checking validity in ClassSetUp causes a crash, where checking in SetUp allows me to mark the test as skipped. That's one of the main points of this PR is to avoid crashes and display errors gracefully.

Unless you have additional thoughts, all my changes here are deliberate.